pandas/core/util/hashing.py

```diff
@@ -205,7 +205,9 @@ def _hash_categorical(c, encoding, hash_key):
     -------
     ndarray of hashed values array, same size as len(c)
     """
-    hashed = hash_array(c.categories.values, encoding, hash_key,
+    # Convert ExtensionArrays to ndarrays
+    values = np.asarray(c.categories.values)
```

Comment:
Same as above. `Categorica[ExtensionArray].values` will be an EA, and this has to be an ndarray.

Reply:
I don't think so, `asarray` won't copy an ndarray, so calling it on an ndarray should be fine.

`is_extension_array_dtype` takes ~ 2 microseconds, and `asarray` takes ~10 ns, so better to just always call asarray instead of checking.

Reply:
do we still need this change? what is it for?

Reply:
I tried removing it, and `test_hash_scalar` failed for `Interval(0, 1)` (I've also parametrized the test):
https://github.com/pandas-dev/pandas/blob/1033e8b1195d4071253889ada60523832285354c/pandas/tests/util/test_hashing.py#L91-L102